### DISCUSSION TITLE
Request/Response interception on Test Request

### DISCUSSION THREAD

**Discussion by willsmith9182** [scalar/scalar#1919] [Q&A] [answered]
Is there anyway to intercept and adjust headers on requests as they are made from the scalar UI?

One use case I have is to attach recaptcha tokens to requests that are generated in the clients browser as they are sent. 

We have (rightly or wrongly) managed to stitch this into a swagger ui currently and im wondering about what my options are if i was considering scalar when we jump to net 9 eventually

**Comment by marclave** [comment]
i have fantastic news @willsmith9182 , we should have something in the works next month that will be the foundation for this :) so yes it will be supported! i will update this thread accordingly

**Comment by willsmith9182** [comment]
Ooof well that is exciting news!

**Comment by willsmith9182** [comment, marked as answer]
```
 <script>
(function(send) {
    XMLHttpRequest.prototype.send = function(data) {
        this.setRequestHeader('Custom-Header', 'HeaderValue');
        console.log('Custom header added -- xmlhttprequest');
        return send.apply(this, arguments);
    };
})(XMLHttpRequest.prototype.send);
</script>
```

this cheeky little devil seemed to do the trick but some first class support would be much appreciated

**Reply by hanspagel** [reply]
Wow, amazing workaround (for now)! :)
